include/deal.II/base/tensor_base.h

```diff
@@ -430,79 +464,85 @@ class Tensor<1,dim,Number>
    * Number.
    */
   template <typename OtherNumber>
-  Tensor<1,dim,Number> &operator = (const Tensor<1,dim,OtherNumber> &rhs);
+  Tensor &operator = (const Tensor<rank_,dim,OtherNumber> &rhs);
 
   /**
    * This operator assigns a scalar to a tensor. To avoid confusion with what
    * exactly it means to assign a scalar value to a tensor, zero is the only
    * value allowed for <tt>d</tt>, allowing the intuitive notation
    * <tt>t=0</tt> to reset all elements of the tensor to zero.
+   *
+   * @relates EnableIfScalar
    */
-  Tensor<1,dim,Number> &operator = (const Number d);
+  template <typename OtherNumber,
+            typename = typename EnableIfScalar<OtherNumber>::type>
+  Tensor<rank_,dim,Number> &operator = (const OtherNumber d);
```

Comment:
template default arguments on functions are C++11 only. But I think you don't even need it -- we don't actually care about the data type, it simply needs to be comparable to zero, i.e., you can do
```
  Assert (d == OtherNumber(), scalar needs to be zero);
  for (i)
    data[i] = Number();   // not the 'd' given by the user, but an equally valid zero
```